devtools/server/actors/replay/module.js

```diff
@@ -1152,18 +1153,31 @@ if (isRecordingOrReplaying) {
       try {
         available = inputStream.available();
       } catch {
-        onResponseEnd();
+        onStreamEnd();
         return;
       }
 
       if (available > 0) {
         const value = NetUtil.readInputStream(inputStream, available);
-        onResponseData(value);
+        onStreamData(value);
+      }
+
+      // For some reason, request streams don't appear to be marked closed properly.
+      // Firefox doesn't support using an actual ReadableStream for fetch bodies,
+      // so every request body SHOULD have all of its data available immediately,
+      // meaning that we can immediately consider all request body streams to
+      // end once they have been read.
```

Comment:
I don't love this, but I also don't really want to dig into internals to figure out of this is expected or a bug.

Reply:
Is there code in the regular devtools net monitor server we can refer to for handling this?

Reply:
It makes this same assumption as far as I could see, and there's nothing that can be reused cleanly.